src/probnum/kernels/_exponentiated_quadratic.py

```diff
@@ -0,0 +1,73 @@
+"""Exponentiated quadratic kernel."""
+
+from typing import Optional
+
+import numpy as np
+import scipy.spatial.distance
+
+import probnum.utils as _utils
+from probnum.type import IntArgType, ScalarArgType
+
+from ._kernel import Kernel
+
+_InputType = np.ndarray
+
+
+class ExpQuad(Kernel[_InputType]):
+    """Exponentiated quadratic / RBF kernel.
+
+    Covariance function defined by :math:`k(x_0, x_1) = \\exp \\big(-\\frac{\\lVert
+    x_0 - x_1 \\rVert^2}{2l^2}\\big)`. This kernel is also known as the squared
+    exponential or radial basis function kernel.
+
+    Parameters
+    ----------
+    input_dim :
+        Input dimension of the kernel.
+    lengthscale
+        Lengthscale of the kernel. Describes the input scale on which the process
+        varies.
+
+    See Also
+    --------
+    RatQuad : Rational quadratic kernel.
+    Matern : Matern kernel.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from probnum.kernels import ExpQuad
+    >>> K = ExpQuad(input_dim=1, lengthscale=0.1)
+    >>> K(np.linspace(0, 1, 3)[:, None])
+    array([[1.00000000e+00, 3.72665317e-06, 1.92874985e-22],
+           [3.72665317e-06, 1.00000000e+00, 3.72665317e-06],
+           [1.92874985e-22, 3.72665317e-06, 1.00000000e+00]])
+    """
+
+    def __init__(self, input_dim: IntArgType, lengthscale: ScalarArgType = 1.0):
+        self.lengthscale = _utils.as_numpy_scalar(lengthscale)
+        super().__init__(input_dim=input_dim, output_dim=1)
+
+    def __call__(self, x0: _InputType, x1: Optional[_InputType] = None) -> np.ndarray:
```

Comment:
What are the  reasons for making x1 optional? Wouldn't it be a more clear communication to user that if they want the matrix corresponding to a single point set, they need to plug in both? Again, not a very strong opinion, but I think it is easier to explain that a kernel takes *two* inputs, instead of explaining that a kernel *usually* takes two inputs.

Reply:
This is mainly for efficiency reasons. If you know, that you are computing the pairwise covariance, this often allows for more efficient computation of the kernel matrix. In particular this also saves you 50% memory.